### ISSUE TITLE
[BUG - Release] Exception Unhandled on latest release

### ISSUE THREAD

**Issue by oscar-wos** [UE4SS-RE/RE-UE4SS#313]
**Branch or Release**
https://github.com/UE4SS-RE/RE-UE4SS/commit/013ff54b86d271c575cfc57f9bc8434a04b502e6

**Game and Engine Version**
Palworld UE 5.1.1

**Describe the bug**
Unhandled exception thrown: read access violation.
**_Val** was 0xD8.

**To Reproduce**
Fresh install of Palworld

>download ue4ss repo
>ran git submodule update --init --recursive
>.\build_auto.bat Release
>complains about missing ue4ss.vcxproj
>open ue4ssMonorepo.sln
>build solution
>copy dwmapi + ue4ss.dll
>copy /assets
>crash

**Expected behavior**
Not crashing

**Screenshots and UE4SS Log**
![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/29248751/8e5c59b9-eb3d-473d-98fe-9d80ad3fa095)

**Desktop (please complete the following information):**
win11
visual studio 2022
cmake version 3.27.9
rustc 1.75.0 (82e1608df 2023-12-21)

**Additional context**
2>LINK : warning LNK4286: symbol '?get_default_devices_ref@DefaultTargets@Output@RC@@SAAEAV?$vector@V?$unique_ptr@VOutputDevice@Output@RC@@U?$default_delete@VOutputDevice@Output@RC@@@std@@@std@@V?$allocator@V?$unique_ptr@VOutputDevice@Output@RC@@U?$default_delete@VOutputDevice@Output@RC@@@std@@@std@@@2@@std@@XZ (public: static class std::vector<class std::unique_ptr<class RC::Output::OutputDevice,struct std::default_delete<class RC::Output::OutputDevice> >,class std::allocator<class std::unique_ptr<class RC::Output::OutputDevice,struct std::default_delete<class RC::Output::OutputDevice> > > > & __cdecl RC::Output::DefaultTargets::get_default_devices_ref(void))' defined in 'DynamicOutput.lib(Output.obj)' is imported by 'ASMHelper.lib(ASMHelper.obj)'
2>UE4SS.vcxproj -> C:\Users\user\Desktop\RE-UE4SS\VS_Solution\Output\CasePreserving__Debug__Win64\UE4SS\bin\UE4SS.dll
2>Done building project "UE4SS.vcxproj".
========== Build: 2 succeeded, 0 failed, 21 up-to-date, 0 skipped ==========
========== Build completed at 11:25 and took 09.413 seconds ==========

**Comment by UE4SS** [comment]
I've pushed an update to the Unreal submodule.
Could you run these commands to update, and then you need to rebuild, and then test again and let me know if you now get an error instead of a crash ?
```
cd deps/first/Unreal
git checkout main
git pull origin main
```

If you do get a crash again, it would be great if you could double click on the UnrealInitializer function in the call stack in VS and also open the `Autos` window and then take a screenshot.
That way I can see exactly where the crash is occurring (because the line number displayed is confusing me) and also the state of variables in the area of the code that's causing the crash.

**Comment by oscar-wos** [comment]
![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/29248751/e24010ea-2849-454f-a491-3f7459bfb12e)
![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/29248751/7af5d883-02f4-4b91-9a5b-f4a024227f4e)

**Comment by oscar-wos** [comment]
console log
https://pastebin.com/cfZbpC7e

**Comment by UE4SS** [comment]
Can you set `HookProcessInternal` and `HookProcessLocalScriptFunction` to 0 in `UE4SS-settings.ini` and see if that helps ?
If it loads successfully, it would be useful if you could dump all objects and properties, which you can do via the button in the GUI or CTRL + J, and upload it somewhere and link it here.

**Comment by oscar-wos** [comment]
I've set both those values to 0, here's the console log: https://pastebin.com/HeV8bp52

When attempting to dump the object values it crashes:
https://file.io/4g7rcdOGmFcp

![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/29248751/fdccf193-76d0-49a8-82fa-45c273ac2468)
![image](https://github.com/UE4SS-RE/RE-UE4SS/assets/29248751/1149c178-e0de-4877-b4db-3c1187bd729d)

**Comment by UE4SS** [comment]
I'm going to need the games exe (in the Win64 directory) as well as your UE4SS.dll and UE4SS.pdb in order to debug with your .dmp file.

**Comment by oscar-wos** [comment]
Here are the files requested: https://file.io/MbukZzUP0A2G

**Comment by UE4SS** [comment]
Thanks.
I'll have a look at it.

**Comment by UE4SS** [comment]
Could you set `FullMemoryDump` to `1` in the `[CrashDump]` section in `UE4SS-settings.ini`, trigger a crash to produce another dmp file and send it ?
The new dmp file might be significantly bigger but the non-full dmp file doesn't contain what I need in order to debug what I suspect is a memory layout problem.

**Comment by oscar-wos** [comment]
Sorry for the delay, awful upload speed https://ufile.io/ae54yrji

**Comment by UE4SS** [comment]
I have a couple of questions:
1. I've noticed that you're compiling with `WITH_CASE_PRESERVING_NAME` defined, are you sure this game uses this option ? It's fairly rare for games to use this option.
2. According to the memory dump, `UObjectBase::NamePrivate` is at offset 0x20, but our default layout for this engine version (5.1) is offset 0x18. Are you using further customizations beyond changes to UE4SS-settings.ini ?

**Comment by oscar-wos** [comment]
I haven't touched any compiler settings, i followed the guide using the .bat scripts

**Comment by UE4SS** [comment]
It might be set to case preserving by default in Visual Studio for some reason.
Can you select `Game__Debug__Win64`, build, and see if that helps ?
![2024-01-28_21-54_VS2022_BuildMode](https://github.com/UE4SS-RE/RE-UE4SS/assets/91646798/76291a2e-75ab-4d42-ba81-89d7bd4321ad)

**Comment by oscar-wos** [comment]
Fantastic it worked! Thanks for taking time to troubleshoot

**Comment by MMALI3287** [comment]
So is the problem fixed now? When the update of 0.1.3 came in palworld I couldn't start the game when UE4SS is installed.

**Comment by UE4SS** [comment]
> So is the problem fixed now? When the update of 0.1.3 came in palworld I couldn't start the game when UE4SS is installed.

This particular problem is different than the problem other people were having.
I believe the fix for the other problem was to use this experimental build: https://github.com/UE4SS-RE/RE-UE4SS/issues/309#issuecomment-1911975127
And also turning off the BetterSellRates mod, and changing some settings as mentioned here: https://github.com/UE4SS-RE/RE-UE4SS/issues/312#issuecomment-1913334053

**Comment by MMALI3287** [comment]
Thank you for your time to solve the issue.

For me changing this did the trick:
GuiConsoleEnabled = 0
